### PR TITLE
Error handler

### PR DESCRIPTION
**Pull request by alex-schneider**

### CODE REVIEW

**Review by malud — DISMISSED** [review]

File: docs/ERRORS.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Especially some access-control errors are expected ones and may be handled on a specific manner.
+To do so every access-control definition of `basic_auth`, `jwt` or `saml2` can define one or multiple
+`error_handler` with a defined error-type label listed below.
+
+Syntax:
+
+```hcl
+# Without label will catch all validation errors related to this access control. 
+error_handler "error-type-a" "error-type-b" {
+  # Behaves like an endpoint
```
Comment (filex) [inline docs/ERRORS.md:31]: this important info should also be in the explaining text (outside of the code snippet)

File: docs/ERRORS.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Those categories will help you to identify the related issues faster.
+
+## Messages
+
+Couper errors distinguish between the synopsis which will be sent to the client, and
+the log message with more error context to prevent accidentally exposing confidential data.
+
+## Access-Control `error_handler`
+
+Especially some access-control errors are expected ones and may be handled on a specific manner.
```
Comment (filex) [inline docs/ERRORS.md:22]:
```suggestion
Especially some access-control errors are expected to occur and may need special handling, e.g. sending a specific response for missing login credentials.
```

File: docs/ERRORS.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+To do so every access-control definition of `basic_auth`, `jwt` or `saml2` can define one or multiple
+`error_handler` with a defined error-type label listed below.
+
+Syntax:
+
+```hcl
+# Without label will catch all validation errors related to this access control.
```
Comment (filex) [inline docs/ERRORS.md:29]: this important info should also be in the explaining text (outside of the code snippet)

File: docs/ERRORS.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  # Behaves like an endpoint
+  response {
+    status = 403
+  }
+}
+```
+
+### Error-Types
+
+| Type                              | Description |
```
Comment (filex) [inline docs/ERRORS.md:40]: should we add a column for the `Standard handling`? "Send error template with 401", "Send error template with 403" …

**Review by filex — CHANGES_REQUESTED** [review]

**Review by malud — APPROVED** [review]

**Pull request merged**